pkgs/sketch_pad/lib/editor/editor.dart

```diff
@@ -173,6 +173,21 @@ class _EditorWidgetState extends State<EditorWidget> implements EditorService {
         (CodeMirror editor, [HintOptions? options]) {
           return options!.results;
         }.toJS);
+
+    // Listen for document body to be visible, then force a code mirror refresh.
+    final observer = web.IntersectionObserver(
```

Comment:
Not for this PR, but I wonder if there isn't some lightweight (extension method based?) wrapper that could make the package:web APIs here slightly more user-friendly?

Reply:
Yeah, a lot of the pain comes from this parameter being typed as `JSFunction`, making it potentially unclear what to specify in the callback. I think this is a pretty common pain point for `package:web` currently.

Once we can update our `package:web` version, we should be able to specify the type parameter for `JSArray` which should help a bit.

Reply:
Since Flutter 3.19 you should be able to update to web: ^0.5.0? Is there any plugin missing?